qa/tasks/scripts/ses_network_failure.sh

```diff
@@ -0,0 +1,245 @@
+    
+    public_interface_list+=($public_interface)
+    cluster_interface_list+=($cluster_interface)
+    
+    minion_list+=($minion)
+
+    case $1 in 
+        delay|delay_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem delay 100ms 10ms distribution normal"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion  cmd.run "tc qdisc add dev $public_interface root netem delay 100ms 10ms distribution normal"
+            fi
+            ;;
+        packet_loss|packet_loss_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem loss 0.3% 25%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem loss 0.3% 25%"
+            fi
+            ;;
+        packet_dup|packet_dup_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem duplicate 1%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem duplicate 1%"
+            fi
+            ;;
+        packet_corruption|packet_corruption_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem corrupt 0.1%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem corrupt 0.1%"
+            fi
+            ;;
+        packet_reordering|packet_reordering_all) salt $minion cmd.run "tc qdisc add dev $cluster_interface root netem delay 10ms reorder 25% 50%"
+            if [ "$public_interface" != "$cluster_interface" ]
+            then
+                salt $minion cmd.run "tc qdisc add dev $public_interface root netem delay 10ms reorder 25% 50%"
+            fi
+            ;;
+    esac
+
+done
+}
+
+remove_netem_rules () {
+case $1 in 
+    delay_all|packet_loss_all|packet_dup_all|packet_corruption_all|packet_reordering_all) simulated_minions=$(echo -e "$2" | wc -l)
+    ;;
+    *) simulated_minions=$2
+    ;;
+esac
+
+for rmqdisc in $(seq 0 $(($simulated_minions-1)))
+do
+    if [ ! -z ${cluster_interface_list[$rmqdisc]} ] && [ "${cluster_interface_list[$rmqdisc]}" != "${public_interface_list[$rmqdisc]}" ]
+    then
+           salt ${minion_list[$rmqdisc]} cmd.run "tc qdisc del dev ${cluster_interface_list[$rmqdisc]} root netem || true"
+    fi
+    
+    salt ${minion_list[$rmqdisc]} cmd.run "tc qdisc del dev ${public_interface_list[$rmqdisc]} root netem || true"
+done
+
+unset cluster_interface_list public_interface_list minion_list
+}
+
+net_storage_minions=$(salt-run select.minions roles=storage | awk '{print $2}')
+net_monitor_minions=$(salt-run select.minions roles=mon | awk '{print $2}')
+
+net_storage_minions_count=$(echo "$net_storage_minions" | wc -l)
```

Comment:
If you're trying to count number of minions in list it is better to do via arrays, like:
```
storage_minions=($(salt-run select.minions roles=mon | awk '{print $2}'))
storage_minions_count=${#net_storage_minions[@]}
```